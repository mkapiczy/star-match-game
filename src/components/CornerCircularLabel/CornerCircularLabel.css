.corner-circular-label {
    position: absolute;
    display: flex;
    align-content: center;
    justify-content: center;
    border-radius: 50%;
    width: 3.5vmin;
    height: 3.5vmin;
    line-height: 3.5vmin;
    text-align: center;
    text-decoration: none;
    background-color: #c300c3;
    font-size: 1.25vh;
    font-weight: bold;
    color: white;
}

.corner-circular-label.top-right{
    right: -1em;
    top: -1em;
}

.corner-circular-label.top-left{
    left: -1em;
    top: -1em;
}

.corner-circular-label.bottom-left{
    left: -1em;
    bottom: -1em
}

.corner-circular-label.bottom-right{
    right: -1em;
    bottom: -1em;
}