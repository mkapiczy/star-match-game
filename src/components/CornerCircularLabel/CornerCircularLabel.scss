.corner-circular-label {
  position: absolute;
  display: flex;
  align-content: center;
  justify-content: center;
  border-radius: 50%;
  width: 3.5vmin;
  height: 3.5vmin;
  line-height: 3.5vmin;
  text-align: center;
  text-decoration: none;
  background-color: #c300c3;
  font-size: 1.25vh;
  font-weight: bold;
  color: white;

  &.top-right {
    right: -1em;
    top: -1em;
  }

  &.top-left {
    left: -1em;
    top: -1em;
  }

  &.bottom-left {
    left: -1em;
    bottom: -1em
  }

  &.bottom-right {
    right: -1em;
    bottom: -1em;
  }
}