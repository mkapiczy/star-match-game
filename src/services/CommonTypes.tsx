export enum Corner {
    TOP_RIGHT,
    TOP_LEFT,
    BOTTOM_RIGHT,
    BOTTOM_LEFT
}

export enum GameState {
    WON,
    LOST,
    IN_PROGRESS,
    NOT_INITIALIZED
}