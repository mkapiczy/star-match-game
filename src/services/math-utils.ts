const random = (min: number, max: number) => Math.floor(Math.random() * (max - min + 1) + min);

const randomSum = (arr: Array<number>, max: number): number => {
    const sets: Array<Array<number>> = [[]];
    const sums: Array<number> = [];
    for (let i = 0; i < arr.length; i++) {
        for (let j = 0, len = sets.length; j < len; j++) {
            const candidateSet = sets[j].concat(arr[i]);
            const candidateSum = candidateSet.reduce((a: number, b: number) => a + b, 0)
            if (candidateSum <= max) {
                sets.push(candidateSet);
                sums.push(candidateSum);
            }
        }
    }
    return sums[random(0, sums.length - 1)];
}

export {random, randomSum}