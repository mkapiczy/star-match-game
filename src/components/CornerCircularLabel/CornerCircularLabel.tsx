import React from 'react';
import './CornerCircularLabel.css';
import {Corner} from '../../services/CommonTypes'

interface Props {
    value: string | number,
    corner: Corner
}

const CornerCircularLabel: React.FC<Props> = (props: Props) => {
    const resolveCornerClass = () => {
        switch (props.corner) {
            case Corner.BOTTOM_RIGHT:
                return 'bottom-right'
            case Corner.BOTTOM_LEFT:
                return 'bottom-left'
            case Corner.TOP_LEFT:
                return 'top-left'
            case Corner.TOP_RIGHT:
                return 'top-right'
            default:
                return ''
        }
    }
    return (
            <div className={"corner-circular-label " + (resolveCornerClass())} onClick={(e) => e.stopPropagation()}>
                {props.value}
            </div>
    )
}


export default CornerCircularLabel;