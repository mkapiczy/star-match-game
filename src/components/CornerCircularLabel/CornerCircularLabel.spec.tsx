import React from 'react';
import CornerCircularLabel from "./CornerCircularLabel";
import renderer from 'react-test-renderer';
import {Corner} from "../../services/CommonTypes";

describe("CornerCircularLabel component snapshot tests", () => {
    it("should match snapshot", () => {
        const component = renderer.create(
            <CornerCircularLabel value={2} corner={Corner.TOP_RIGHT}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})