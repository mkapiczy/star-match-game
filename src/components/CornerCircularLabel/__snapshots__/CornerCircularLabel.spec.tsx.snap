// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CornerCircularLabel component snapshot tests should match snapshot 1`] = `
<div
  className="corner-circular-label top-right"
  onClick={[Function]}
>
  2
</div>
`;
